src/Symfony/Bundle/SecurityBundle/Templating/Helper/LogoutUrlHelper.php

```diff
@@ -97,6 +111,14 @@ public function getLogoutUrl($key)
      */
     private function generateLogoutUrl($key, $referenceType)
     {
+        // Fetch the current provider key from token, if possible
+        if (null === $key && $this->tokenStorage) {
+            $token = $this->tokenStorage->getToken();
+            if ($token && method_exists($token, 'getProviderKey')) {
+                $key = $token->getProviderKey();
+            }
+        }
+
         if (!array_key_exists($key, $this->listeners)) {
             throw new \InvalidArgumentException(sprintf('No LogoutListener found for firewall key "%s".', $key));
```

Comment:
I do understand the change, but if the `getProviderKey()` method does not exist, the `$key` will still be `null` here, in which case, the error message is not the right one.